src/api/app/components/user_avatar_component.rb

```diff
@@ -0,0 +1,21 @@
+class UserAvatarComponent < ApplicationComponent
+  def initialize(user)
+    super
+
+    @user = user
+  end
+
+  private
+
+  def avatar_object
+    @avatar_object = User.find_by_login(@user)
```

Comment:
This could be memoized, so it is not fetch over and over again.

```suggestion
    @avatar_object ||= User.find_by_login(@user)
```